game-core/src/main/java/games/strategy/triplea/settings/GameSetting.java

```diff
@@ -60,4 +61,15 @@ default T getValueOrThrow() {
    * Unregisters {@code listener} to no longer receive a notification whenever the setting value has changed.
    */
   void removeListener(Consumer<GameSetting<T>> listener);
+
+
+
+  static String getFromSetting(final GameSetting<char[]> setting) {
```

Comment:
Now that this method is in `GameSetting`, the `Setting` suffix in the name seems redundant.  Looking at the body of the method, a more appropriate name might be `getValueAsStringOrThrow()`, but that might be considered too busy. :smile:

Looking at the usage of this method once again, it appears it's not even really necessary.  You could use the existing `Arrays#withSensitiveArrayAndReturn()` method.  For example, in `IEmailSender`:

```diff
diff --git a/game-core/src/main/java/games/strategy/engine/pbem/IEmailSender.java b/game-core/src/main/java/games/strategy/engine/pbem/IEmailSender.java
index 3698afed2..6dceac041 100644
--- a/game-core/src/main/java/games/strategy/engine/pbem/IEmailSender.java
+++ b/game-core/src/main/java/games/strategy/engine/pbem/IEmailSender.java
@@ -1,5 +1,7 @@
 package games.strategy.engine.pbem;
 
+import static org.triplea.common.util.Arrays.withSensitiveArrayAndReturn;
+
 import java.io.File;
 import java.io.IOException;
 
@@ -7,7 +9,6 @@ import javax.annotation.Nonnull;
 import javax.annotation.concurrent.Immutable;
 
 import games.strategy.triplea.settings.ClientSetting;
-import games.strategy.triplea.settings.GameSetting;
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 
@@ -70,8 +71,8 @@ public interface IEmailSender {
             ClientSetting.emailServerPort.getValueOrThrow(),
             ClientSetting.emailServerSecurity.getValueOrThrow()
         ),
-        GameSetting.getFromSetting(ClientSetting.emailUsername),
-        GameSetting.getFromSetting(ClientSetting.emailPassword),
+        withSensitiveArrayAndReturn(ClientSetting.emailUsername::getValueOrThrow, String::new),
+        withSensitiveArrayAndReturn(ClientSetting.emailPassword::getValueOrThrow, String::new),
         subjectPrefix,
         toAddress);
   }
```

If you don't like having to provide the `String::new` method reference in the eight locations this method is used, a new `Arrays` helper method would be appropriate to add:

```java
  public static String convertSensitiveArrayToString(final Supplier<char[]> supplier) {
    return withSensitiveArrayAndReturn(supplier, String::new);
  }
```